production/ksonnet/loki/bloom-compactor.libsonnet

```diff
@@ -0,0 +1,118 @@
+{
+  local k = import 'ksonnet-util/kausal.libsonnet',
+  local container = k.core.v1.container,
+  local containerPort = k.core.v1.containerPort,
+  local pvc = k.core.v1.persistentVolumeClaim,
+  local service = k.core.v1.service,
+  local statefulSet = k.apps.v1.statefulSet,
+  local volume = k.core.v1.volume,
+  local volumeMount = k.core.v1.volumeMount,
+
+  local name = 'bloom-compactor',
+
+  _config+:: {
+    bloom_compactor+: {
+      pvc_size: '256Gi',  // error 'define bloom_compactor.pvc_size',
+      pvc_class: 'fast',  // error 'define bloom_compactor.pvc_class',
+    },
+    loki+: {
+      bloom_compactor+: {
+        enabled: $._config.use_bloom_filters,
+        working_directory: '/data/blooms',
+        compaction_interval: '15m',
+        max_compaction_parallelism: 1,
+      },
+    },
+  },
+
+  local cfg = self._config.bloom_compactor,
+
+  local volumeName = name + '-data',
+  local volumeMounts = [volumeMount.new(volumeName, '/data')],
+
+  bloom_compactor_args::
+    if $._config.use_bloom_filters
+    then
+      $._config.commonArgs {
+        target: 'bloom-compactor',
+      }
+    else {},
+
+  bloom_compactor_ports:: [
+    containerPort.new(name='http-metrics', port=$._config.http_listen_port),
+    containerPort.new(name='grpc', port=9095),
+  ],
+
+  bloom_compactor_data_pvc::
+    if $._config.use_bloom_filters
+    then
+      pvc.new(volumeName)
+      // set disk size
+      + pvc.mixin.spec.resources.withRequests({ storage: $._config.bloom_compactor.pvc_size })
+      // mount the volume as read-write by a single node
+      + pvc.mixin.spec.withAccessModes(['ReadWriteOnce'])
+      // set persistent volume storage class
+      + pvc.mixin.spec.withStorageClassName($._config.bloom_compactor.pvc_class)
+    else {},
+
+
+  bloom_compactor_container::
+    if $._config.use_bloom_filters
+    then
+      container.new(name, $._images.bloom_compactor)
+      // add default ports
+      + container.withPorts($.bloom_compactor_ports)
+      // add target specific CLI arguments
+      + container.withArgsMixin(k.util.mapToFlags($.bloom_compactor_args))
+      // mount the data pvc at given mountpoint
+      + container.withVolumeMountsMixin(volumeMounts)
+      // add globale environment variables
+      + container.withEnvMixin($._config.commonEnvs)
+      // add HTTP readiness probe
+      + container.mixin.readinessProbe.httpGet.withPath('/ready')
+      + container.mixin.readinessProbe.httpGet.withPort($._config.http_listen_port)
+      + container.mixin.readinessProbe.withTimeoutSeconds(1)
+      // define container resource requests
+      + k.util.resourcesRequests('2', '4Gi')
+      // define container resource limits
+      + k.util.resourcesLimits(null, '8Gi')
+    else {},
+
+  bloom_compactor_statefulset:
+    if $._config.use_bloom_filters
+    then
+      statefulSet.new(name, 3, [$.bloom_compactor_container], $.bloom_compactor_data_pvc)
```

Comment:
Can we extract number of replicas into config (or at least a local var above) like you did for bloom-gw below? it will make it more readable/findable